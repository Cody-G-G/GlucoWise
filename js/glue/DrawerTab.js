'use strict';
import React, {Component} from 'react';
import {Icon} from 'native-base';
import {TouchableOpacity, Text} from 'react-native';
import styles from "./styles";

export default class DrawerTab extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <TouchableOpacity onPress={this.props.onPress} style={styles.tabButton}>
                <Icon theme={{iconFamily: "MaterialIcons"}} style={styles.icon} name={this.props.icon}/>

                {(typeof this.props.children) === 'undefined' ?
                    <Text style={styles.tabText}>{this.props.text}</Text> : this.props.children}
            </TouchableOpacity>
        );
    }
}