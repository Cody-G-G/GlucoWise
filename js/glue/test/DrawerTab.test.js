'use strict';
import React from 'react';
import 'react-native';
import {Text} from 'react-native';
import renderer from 'react-test-renderer';
import DrawerTab from "../DrawerTab";

test('DrawerTab_withoutChildren - renders correctly', () => {
    const icon = 'bluetooth';
    const text = "PROP_TEXT";

    const tree = renderer.create(
        <DrawerTab icon={icon}
                   text={text}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('DrawerTab_withChildren - renders correctly', () => {
    const icon = 'bluetooth';
    const text = "PROP_TEXT";

    const tree = renderer.create(
        <DrawerTab icon={icon}>
            <Text>CHILD_TEXT</Text>
            <Text>CHILD_TEXT2</Text>
        </DrawerTab>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});