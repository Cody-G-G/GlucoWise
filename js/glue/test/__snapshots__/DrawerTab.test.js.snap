// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerTab_withChildren - renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "whitesmoke",
      "borderColor": "royalblue",
      "borderWidth": 1.5,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "royalblue",
          "fontSize": 45,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    theme={
      Object {
        "iconFamily": "MaterialIcons",
      }
    }
  >
    
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      CHILD_TEXT
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      CHILD_TEXT2
    </Text>
  </View>
</View>
`;

exports[`DrawerTab_withoutChildren - renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "whitesmoke",
      "borderColor": "royalblue",
      "borderWidth": 1.5,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "royalblue",
          "fontSize": 45,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
    theme={
      Object {
        "iconFamily": "MaterialIcons",
      }
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "royalblue",
        "fontSize": 25,
        "fontWeight": "bold",
      }
    }
  >
    PROP_TEXT
  </Text>
</View>
`;
